chat-server/server.js

```diff
@@ -21,8 +23,68 @@ const welcomeMessage = {
 //We will start with one message in the array.
 const messages = [welcomeMessage];
 
-app.get("/", (request, response) => {
-  response.sendFile(__dirname + "/index.html");
+app.get("/", (req, res) => {
+  res.sendFile(__dirname + "/index.html");
+});
+
+app.post("/messages", (req, res) => {
+  const newMessage = req.body;
+  if (
+    !newMessage.hasOwnProperty("text") ||
+    !newMessage.hasOwnProperty("from") ||
+    newMessage.from === "" ||
+    newMessage.text === "" ||
+    typeof newMessage.text != "string" ||
+    typeof newMessage.from != "string"
+  ) {
+    res.sendStatus(400).send("wrong input");
+  }
+  newMessage["id"] = messages[messages.length - 1].id + 1;
+  newMessage["timeSent"] = new Date();
+  messages.push(newMessage);
+  res.json(messages);
+});
+
+app.get("/messages", (req, res) => {
+  res.json(messages);
+});
+
+app.get("/messages/id/:id", (req, res) => {
```

Comment:
There is no right and wrong, but you generally don't need to include the word 'id' in the route path. you can simply use `/messages/:id`. It makes the endpoint more concise and follows a common convention in **RESTful APIs.**

Reply:
I really had a bad time fixing this 😆 when I tried to create another route (ex: `/message/search`) it was getting recognised as `:id` parameter. I couldn't solve the problem and found this solution. After seeing this message spent an hour but still couldn't solve 😆